File: stringcalculator/java/src/main/java/StringCalculator.java
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class StringCalculator {
    public int add(String stringInput) {
        if (stringInput.equals("")) {
            return 0;
        }
        if (stringInput.endsWith(",")) {
            throw new IllegalArgumentException("Invalid Terminator Expression");
        }

        String delimiterRegExp = "[;,\\n]";
        List<Integer> numbers = Arrays.stream(stringInput.split(delimiterRegExp))
                                      .map(Integer::parseInt)
                                      .filter(num -> num < 1000)
                                      .collect(Collectors.toList());

        List<Integer> negatives = numbers.stream()
                                         .filter(num -> num < 0)
                                         .collect(Collectors.toList());
        if (!negatives.isEmpty()) {
            throw new IllegalArgumentException("Negative Number Error");
        }

        return numbers.stream().mapToInt(Integer::intValue).sum();
    }
}

File: stringcalculator/ts/src/stringCalculator.ts
export class StringCalculator {
  add(stringInput: string) {
    if (stringInput === "") {
      return 0;
    }
    if (stringInput.endsWith(",")) {
      throw new Error("Invalid Terminator Expression");
    }

    const delimiterRegExp = /[;,\n]/;
    let numbers = stringInput
      .split(delimiterRegExp)
      .map((n) => parseInt(n))
      .filter((num) => num < 1000);

    let negatives = numbers.filter((num) => num < 0);
    if (negatives.length > 0) {
      throw new Error("Negative Number Error");
    }

    let finalSum = numbers.reduce((a, c) => a + c, 0);
    return finalSum;
  }
}
